tests/multi_ptr/multi_ptr_convert_assignment_ops.h

```diff
@@ -0,0 +1,124 @@
+
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides code for multi_ptr convert assignment operators
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_CONVERT_ASSIGN_OPS_H
+#define __SYCLCTS_TESTS_MULTI_PTR_CONVERT_ASSIGN_OPS_H
+
+#include "../common/common.h"
+
+#include "../common/section_name_builder.h"
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_convert_assignment_ops {
+
+constexpr int expected_val = 42;
+
+template <typename T, typename SrcAddrSpaceT, typename SrcIsDecorated,
+          typename DstIsDecorated>
+class run_convert_assignment_operators_tests {
+  static constexpr sycl::access::address_space src_space = SrcAddrSpaceT::value;
+  static constexpr sycl::access::decorated src_decorated =
+      SrcIsDecorated::value;
+  static constexpr sycl::access::decorated dst_decorated =
+      DstIsDecorated::value;
+  using src_multi_ptr_t = sycl::multi_ptr<T, src_space, src_decorated>;
+  using dst_multi_ptr_t =
+      sycl::multi_ptr<T, sycl::access::address_space::generic_space,
+                      src_decorated>;
+
+ public:
+  void operator()(const std::string &type_name,
+                  const std::string &src_address_space_name,
+                  const std::string &src_is_decorated_name,
+                  const std::string &dst_is_decorated_name) {
+    auto queue = sycl_cts::util::get_cts_object::queue();
+    T value = user_def_types::get_init_value_helper<T>(expected_val);
+    SECTION(
+        section_name(
+            "Check &operator=(const multi_ptr<value_type, ASP, IsDecorated>&)")
+            .with("T", type_name)
+            .with("src address_space", src_address_space_name)
+            .with("src decorated", src_is_decorated_name)
+            .with("dst address_space", "access::address_space::generic_space")
+            .with("dst decorated", dst_is_decorated_name)
+            .create()) {
+      bool res = false;
+      {
+        sycl::buffer<bool> res_buf(&res, sycl::range<1>(1));
+        sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
+        queue.submit([&](sycl::handler &cgh) {
+          auto res_acc =
+              res_buf.template get_access<sycl::access_mode::write>(cgh);
+          auto acc_for_mptr =
+              val_buffer.template get_access<sycl::access_mode::read>(cgh);
+          cgh.single_task([=] {
+            const src_multi_ptr_t mptr_in(acc_for_mptr);
```

Comment:
I don't get what is happening here.
`src_multi_ptr_t` is supposed to iterate all address spaces. Is that right? If so, can we initialize a multi_ptr to local address space with an accessor to buffer allocated in a global address space?

Reply:
Yes, you are right
This comment has been applied in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/349/commits/ecd3ef4080b0903520ac7f9b00e353c175898af6